### PR TITLE
VIH-8009 Video Web iPad Choose camera and mic option errors

### PR DESCRIPTION
**Pull request by ZJianWang**
### JIRA link (if applicable) ###
VIH-8009

### Change description ###
- Get rid of the two lines of this.pexipAPI.video_source and this.pexipAPI.audio_source 
- Use renegotiating rather then disconnecting and reconnecting
- Stopped multiple cameras from been accessed at same time to fix iOS/Mac/Safari issues
- Select device component is now the outgoing stream to pexip
- Reworked self test to behave the same as the other reworked components
- Fixed bugs in has multiple device and another observable that would only emit once

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts [hunk 1]
```diff
@@ -34,17 +34,34 @@ export class SelectMediaDevicesComponent implements OnInit, OnDestroy {
...
         private profileService: ProfileService,
         private videoFilterService: VideoFilterService
     ) {}
 
     ngOnInit() {
-        this.userMediaService.connectedDevices$.pipe(takeUntil(this.destroyedSubject)).subscribe(connectedDevices => {
-            this.availableCameraDevices = connectedDevices.filter(device => device.kind === 'videoinput');
-            this.availableMicrophoneDevices = connectedDevices.filter(device => device.kind === 'audioinput');
+        this.userMediaService.connectedVideoDevices$.pipe(takeUntil(this.destroyedSubject)).subscribe(cameraDevices => {
+            this.availableCameraDevices = null;
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts:46]: why do you set to null and then the new incoming variable?

File: VideoWeb/VideoWeb/ClientApp/src/app/services/user-media.service.ts [hunk 2]
```diff
@@ -231,17 +236,15 @@ export class UserMediaService {
     }
 
     hasMultipleDevices(): Observable<boolean> {
-        return zip(this.connectedVideoDevices, this.connectedMicrophoneDevices).pipe(
-            take(1),
+        return zip(this.connectedVideoDevices$, this.connectedMicrophoneDevices$).pipe(
             map(deviceList => {
-                return deviceList[0].length > 1 || deviceList[1].length > 1;
+                return deviceList[0].length > 1; // || deviceList[1].length > 1;
```
Comment (kevin-hmcts) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/user-media.service.ts:241]: Consider removing this comment

**Review by kevin-hmcts — DISMISSED** [review]
LGTM

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts (hunk 1 above)
Comment (robertbennett1998) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts:46]: Left over code from something else. Removed it.

**Review by kevin-hmcts — APPROVED** [review]
LGTM

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**